ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs

```diff
@@ -159,3 +179,61 @@ impl IndexerStream for DatastreamServer {
         ))
     }
 }
+
+/// Builds the response for the raw datastream request. Partial batch is ok, i.e., a batch with transactions < 1000.
+fn raw_datastream_response_builder(
+    data: Vec<EncodedTransactionWithVersion>,
+    chain_id: u32,
+) -> RawDatastreamResponse {
+    RawDatastreamResponse {
+        response: Some(DatastreamProtoResponse::Data(TransactionsOutput {
+            transactions: data
+                .into_iter()
+                .map(|(encoded, version)| TransactionOutput {
+                    encoded_proto_data: encoded,
+                    version,
+                    ..TransactionOutput::default()
+                })
+                .collect(),
+        })),
+        chain_id,
+    }
+}
+
+/// Fetches data from cache or the file store. It returns the data if it is ready in the cache or file store.
+/// Otherwise, it returns the status of the data fetching.
+async fn data_fetch(
```

Comment:
comment that if there are errors we will retry.

Reply:
done.